src/notation/qml/MuseScore/NotationScene/internal/HarpPedalPopup.qml

```diff
@@ -0,0 +1,242 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2022 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import MuseScore.Ui 1.0
+import MuseScore.UiComponents 1.0
+import MuseScore.NotationScene 1.0
+
+StyledPopupView {
+    id: root
+
+    HarpPedalPopupModel {
+        id: harpModel
+
+        onIsDiagramChanged: updatePosition(harpModel.pos, harpModel.size)
+    }
+
+    property QtObject model: harpModel
+
+    property variant pedalState: harpModel.pedalState
+
+    contentWidth: menuItems.width
+
+    contentHeight: menuItems.height
+
+    margins: 0
+
+    closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutside
+
+    showArrow: false
+
+    function updatePosition(pos, size) {
+
+        // Default: open above position of diagram
+        setOpensUpward(true)
+        root.x = (pos.x + size.x / 2) - contentWidth / 2
+        root.y = pos.y - size.y - contentHeight
+
+
+        // For diagrams below stave, position above stave to not obscure it
+        if (harpModel.belowStave) {
+            root.y = harpModel.staffPos.y - size.y - contentHeight
+        }
+
+        // not enough room on window to open above so open below stave
+        var globPos = mapToItem(notationView.windowContent, Qt.point(root.x, root.y))
+        if (globPos.y < 0) {
+            setOpensUpward(false)
+            root.y = harpModel.staffPos.y + harpModel.staffHeight
+        }
+
+        // not enough room below stave to open so open above
+        if (root.y > notationView.height) {
+            root.y = pos.y - size.y
+        }
+    }
+
+    ColumnLayout {
+        id: menuItems
+
+        RowLayout {
```

Comment:
Would it be possible to refactor this into a GridLayout? I think that could result in cleaner code. Also, for the dummy items: it would be preferable to solve this using Layout.leftMargin and Layout.rightMargin, but if you really need dummy items, then I think you can just use an `Item` rather than a dummy ColumnLayout.

Reply:
Including the diagram/text buttons in the grid?  I remember running into trouble with using the separator line in the grid too.